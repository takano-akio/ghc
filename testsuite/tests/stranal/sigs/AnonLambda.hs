module AnonLambda where

g :: Int -> Bool
{-# NOINLINE g #-}
g = (==0)

-- This test ensures that the CPR property of the anonymous lambda
-- Does not escape to f (which has arity 1)

f = \x -> if g x then \y -> x + y + 1
                 else \y -> x + y + 2
