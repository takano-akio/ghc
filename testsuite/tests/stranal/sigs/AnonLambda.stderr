
==================== Strictness signatures ====================
AnonLambda.$trModule: tm(tm1,tm1)
AnonLambda.f: <S(S),U(U)>
AnonLambda.g: <S(S),1*U(1*U)>



==================== Strictness signatures ====================
AnonLambda.$trModule: tm(tm1,tm1)
AnonLambda.f: <S(S),1*U(U)>
AnonLambda.g: <S(S),1*U(1*U)>


